src/Components/Modal/Modal.jsx

```diff
@@ -0,0 +1,60 @@
+import React, { useCallback, useEffect, useRef } from "react";
+import CloseIcon from "@material-ui/icons/Close";
+
+import "./Modal.scss";
+
+const Modal = ({ title, onClose, open = false, renderClose = true, closeOnClickOutside = true, children }) => {
+  const ref = useRef(null);
+
+  const escapeListener = useCallback(
+    (e) => {
+      if (e.key === "Escape") {
+        onClose?.();
+      }
+    },
+    // Modal specific depenency
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  );
+
+  const clickListener = useCallback(
+    (e) => {
+      if (!ref.current?.contains(e.target)) {
+        onClose?.();
+      }
+    },
+    // Modal specific depenency
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [ref.current]
```

Comment:
I think you can't use a ref as a dependency.

Reply:
your right!